pkg/tanka/load.go

```diff
@@ -0,0 +1,93 @@
+package tanka
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/grafana/tanka/pkg/jsonnet/jpath"
+	"github.com/grafana/tanka/pkg/kubernetes"
+	"github.com/grafana/tanka/pkg/kubernetes/manifest"
+	"github.com/grafana/tanka/pkg/process"
+	"github.com/grafana/tanka/pkg/spec"
+	"github.com/grafana/tanka/pkg/spec/v1alpha1"
+	"github.com/pkg/errors"
+)
+
+// Load loads the Environment at `path`. It automatically detects whether to
+// load inline or statically
+func Load(path string, opts Opts) (*LoadResult, error) {
+	_, base, err := jpath.Dirs(path)
+	if err != nil {
+		return nil, err
+	}
+
+	loader, err := chooseLoader(base)
+	if err != nil {
+		return nil, err
+	}
+
+	env, err := loader.Load(path, opts.JsonnetOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := checkVersion(env.Spec.ExpectVersions.Tanka); err != nil {
+		return nil, err
+	}
+
+	processed, err := process.Process(env.Data, *env, opts.Filters)
+	if err != nil {
+		return nil, err
+	}
+
+	return &LoadResult{Env: env, Resources: processed}, nil
+}
+
+// chooseLoader detects whether the environment is inline or static and picks
+// the approriate loader
+func chooseLoader(base string) (Loader, error) {
```

Comment:
Nit: somehow `detectLoader` sounds better to me, because we are not explicitly choosing one but rather detect which one we should use based on some _environmental_ conditions.